src/components/views/WaitForEcashPayment.tsx

```diff
@@ -43,47 +40,27 @@ const WaitForEcashPayment = ({ request, onSuccess }: WaitForEcashPaymentProps) =
       setAmountData(null);
    }
 
-   const pollPayment = useCallback(async () => {
-      console.log('checking for payment', request.id);
-      const { paid, token } = await checkPaymentRequest(request.id);
-      if (paid) {
+   const checkPayment = useCallback(async () => {
+      const { token } = await checkPaymentRequest(request.id);
+      if (token) {
          await handleClaimToSourceMint(token);
          onSuccess();
-         handleClose();
-         console.log('Payment completed', token);
-         if (pollIntervalRef.current) clearInterval(pollIntervalRef.current);
-      } else {
-         pollCountRef.current++;
-         if (pollCountRef.current >= MAX_POLL_COUNT) {
-            if (pollIntervalRef.current) clearInterval(pollIntervalRef.current);
-            setPrTimeout(true);
-         }
+         return true;
       }
-   }, [checkPaymentRequest, request.id, onSuccess]);
-
-   const startPolling = useCallback(() => {
-      setPrTimeout(false);
-      pollCountRef.current = 0;
-      pollPayment();
+      return false;
+   }, [checkPaymentRequest, request.id, onSuccess, handleClaimToSourceMint]);
 
-      // Clear any existing interval
-      if (pollIntervalRef.current) clearInterval(pollIntervalRef.current);
+   const { start, stop, isPolling } = usePolling(checkPayment, 5000, 6000);
 
-      pollIntervalRef.current = setInterval(pollPayment, 5000); // Poll every 5 seconds
-   }, [pollPayment]);
+   const checkAgain = async () => {
+      await checkPayment();
+   };
 
+   /* Start polling when component mounts */
    useEffect(() => {
```

Comment:
I am not sure I see a value in starting and stoping like this. All you need to do is to set the interval value with useInterval to some number to start the interval and change it to null to stop it. You could do that without custom hook like this:
```
...
const [pollingPeriod, setPollingPeriod] = useState<number | null>(5000);
...
useInterval(checkPayment, pollingPeriod);

useTimeout(() => {
  setPollingPeriod(null)
}, 6000);
```

if you see this is a repeating pattern makes sense to abstract it into custom usePolling hook but then this hook can look like:

```
const usePolling = (callback: Function, delay?: number | null, timeout?: number | null) => {
    const [isReady, _, reset] = useTimeout(() => {
        setPollingPeriod(null)
    }, timeout);
    
    const pollingPeriod = !isReady() ? delay : null;
    
    useInterval(callback, pollingPeriod);
    
    return {
        isPolling: !!pollingPeriod,
        reset
    };
}
```

Reply:
Yea this is much cleaner, I think I over complicated it because I didn't understand how to use useTimeout.

Keeping the hook because I'll use polling elsewhere (receive lightning payment). Please check again.